### PR TITLE
Implement `dbStats` for SQLite

### LINKED ISSUE #3259: Implement `dbStats` and `collStats` for SQLite
### What should be done?

TBD

### Where?

TBD

### Definition of Done

- all handlers updated;
- unit tests added/updated;
- integration/compatibility tests added/updated;
- spot refactorings done;
- user documentation updated or an issue to create documentation created;
- something else?

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #3259.

It does not cover `collStats`, that will be in another PR.
In this issue we are not adding new integration tests, we want SQLite supported sooner, you'll notice codecov warning for untested errors. We will test them when we use universal handler in a few weeks time.

## Readiness checklist

- [x] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/backends/sqlite/database.go [hunk 1]
```diff
@@ -101,6 +101,35 @@ func (db *database) RenameCollection(ctx context.Context, params *backends.Renam
...
+	// Call ANALYZE to update statistics, the actual statistics are needed to estimate the number of rows in all tables,
+	// see https://www.sqlite.org/lang_analyze.html.
+	q := `ANALYZE`
+	if _, err := d.ExecContext(ctx, q); err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	// Total size is the disk space used by the database.
+	// See https://www.sqlite.org/pragma.html#pragma_page_size.
+	q = `PRAGMA PAGE_SIZE`
```
Comment (AlekSi) [inline internal/backends/sqlite/database.go:122]: Let's use https://www.sqlite.org/dbstat.html /cc @rumyantseva

Comment (chilagrow) [reply internal/backends/sqlite/database.go:122]: Sounds good :+1:

**Comment by chilagrow** [comment]
Response of implementation
```
test> db.runCommand({dbStats: 1})                                                                                                                                                                   [26/510]
{                                                                                                                                                                                                           
  db: 'test',                                                                                                                                                                                               
  collections: Long("0"),                                                                                                                                                                                   
  views: 0,                                                                                                                                                                                                 
  objects: Long("0"),                                                                                                                                                                                       
  dataSize: Long("0"),                                                                                                                                                                                      
  storageSize: Long("0"),                                                                                                                                                                                   
  indexes: Long("0"),                                                                                                                                                                                       
  indexSize: Long("0"),                                                                                                                                                                                     
  totalSize: Long("0"),                                                                                                                                                                                     
  scaleFactor: 1,                                                                                                                                                                                           
  ok: 1                                                                                                                                                                                                     
}                                                                                                                                                                                                           
test> db.coll.insertMany([{_id: 1}, {_id: 2}])                                                                                                                                                              
{ acknowledged: true, insertedIds: { '0': 1, '1': 2 } }                                                                                                                                                     
test> db.runCommand({dbStats: 1})                                                                                                                                                                           
{                                                                                                                                                                                                           
  db: 'test',                                                                                                                                                                                               
  collections: Long("1"),                                                                                                                                                                                   
  views: 0,                                                                                                                                                                                                 
  objects: Long("2"),                                                                                                                                                                                       
  avgObjSize: Long("2048"),                                                                                                                                                                                 
  dataSize: Long("4096"),                                                                                                                                                                                   
  storageSize: Long("4096"),                                                                                                                                                                                
  indexes: Long("1"),                                                                                                                                                                                       
  indexSize: Long("4096"),                                                                                                                                                                                  
  totalSize: Long("32768"),                        
  scaleFactor: 1,                                  
  ok: 1                                            
}                                                  
```

**Comment by chilagrow** [comment]
Example of `dbstat` and `sqlite_schema` table contents, with empty collection `new-collection` and 2 documents collection `coll`.
```
sqlite> select name, pgsize, ncell from dbstat;
sqlite_schema|4096|8
_ferretdb_collections|4096|2
sqlite_autoindex__ferretdb_collections_1|4096|2
sqlite_autoindex__ferretdb_collections_2|4096|2
new-collection_82bca3b8|4096|0
new-collection_82bca3b8_id|4096|0
sqlite_stat1|4096|2
coll_f244e9a7|4096|2
coll_f244e9a7_id|4096|2
```
```
sqlite> select type, name, tbl_name from sqlite_schema;
table|_ferretdb_collections|_ferretdb_collections
index|sqlite_autoindex__ferretdb_collections_1|_ferretdb_collections
index|sqlite_autoindex__ferretdb_collections_2|_ferretdb_collections
table|new-collection_82bca3b8|new-collection_82bca3b8
index|new-collection_82bca3b8_id|new-collection_82bca3b8
table|sqlite_stat1|sqlite_stat1
table|coll_f244e9a7|coll_f244e9a7
index|coll_f244e9a7_id|coll_f244e9a7
```

File: internal/backends/sqlite/database.go [hunk 2]
```diff
@@ -102,8 +102,75 @@ func (db *database) RenameCollection(ctx context.Context, params *backends.Renam
...
+	// See https://www.sqlite.org/schematab.html and
+	// also https://www.sqlite.org/fileformat.html.
+	q = `
+		SELECT
+		    COUNT(s.name)             AS CountTables,
+		    COALESCE(SUM(d.pgsize),0) AS SizeTables,
+		    COALESCE(SUM(d.ncell),0)  AS CountCells
+		FROM sqlite_schema AS s
+			LEFT JOIN dbstat AS d ON d.name = s.tbl_name
+		WHERE s.type = 'table' AND s.name NOT LIKE ? AND s.name <> ?`
```
Comment (AlekSi) [inline internal/backends/sqlite/database.go:145]: Another option is to get a list of collections/tables (see #3292) and use `d.name IN (?, ?, ?, ...)`. After all, the registry is responsible for maintaining of collections/tables, not `sqlite_schema`

Comment (chilagrow) [reply internal/backends/sqlite/database.go:145]: That is a really good point. There's another advantage of doing that way, I think I can use sqlite aggregation query if I avoid joining with sqlite_schema table. Plus registry should give me the number of tables without querying!

**Review by rumyantseva — DISMISSED** [review]
The code looks reasonable to me

**Review by rumyantseva — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**